server/session/handler_book_edit.go

```diff
@@ -0,0 +1,70 @@
+package session
+
+import (
+	"fmt"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/sandertv/gophertunnel/minecraft/protocol/packet"
+	"golang.org/x/exp/slices"
+)
+
+type BookEditHandler struct{}
+
+// Handle ...
+func (b BookEditHandler) Handle(p packet.Packet, s *Session) error {
+	pk := p.(*packet.BookEdit)
+	it, err := s.inv.Item(int(pk.InventorySlot))
+	if err != nil {
+		return fmt.Errorf("invalid inventory slot index %v", pk.InventorySlot)
+	}
+	book, ok := it.Item().(item.WritableBook)
+	if !ok {
+		return fmt.Errorf("inventory slot %v does not contain a writable book", pk.InventorySlot)
+	}
+	// check page number and text beforehand to reduce repetition, shouldn't matter as the default values
+	// match these constraints
+	page := uint(pk.PageNumber)
+	if page >= 50 {
+		return fmt.Errorf("page number %v is out of bounds", pk.PageNumber)
+	}
+	if len(pk.Text) > 256 {
+		return fmt.Errorf("text can not be longer than 256 bytes")
+	}
```

Comment:
This logic does not match the logic in (WriteableBook).Set()

Reply:
I'm pretty sure it does,
`
if len(text) > 256 {
	panic("text longer then 256 bytes")
}
`